core/opendaq/signal/src/connection_impl.cpp

```diff
@@ -16,6 +16,13 @@ ErrCode ConnectionImpl::enqueue(IPacket* packet)
 {
     OPENDAQ_PARAM_NOT_NULL(packet);
 
+    if (!port.getActive())
+    {
+        void* intf;
+        if (OPENDAQ_FAILED(packet->borrowInterface(IEventPacket::Id, &intf)))
```

Comment:
for packets we have IPacket.getPacketType() which should be more optimal than borrowing